modules/yahoosspBidAdapter.js

```diff
@@ -277,21 +277,29 @@ function generateOpenRtbObject(bidderRequest, bid) {
       }
     } else {
       outBoundBidRequest.site.id = bid.params.dcn;
-    };
+    }
+
+    if (bidderRequest.gppConsent) {
+      outBoundBidRequest.regs.ext.gpp = bidderRequest.gppConsent.gppString;
+      outBoundBidRequest.regs.ext.gpp_sid = bidderRequest.gppConsent.applicableSections;
+    } else if (bidderRequest.ortb2?.regs?.gpp) {
```

Comment:
why doesnt aol have this same logic?

Reply:
the aol bidder adapter is the old one we are migrating off. Also there we have undocumented support for openRTB config, we don't want to add this support in the legacy adapter.

Reply:
if `bidderRequest.gppConsent` is defined, then the `ortb2.regs` properties are as well; you can save some bytes and complexity by looking only at the latter. 

In fact, you could replace a lot of the code in here (esp. the giant `appendFirstPartyData` function) with just `mergeDeep(outBoundBidRequest, bidderRequest.ortb2)`.

Reply:
https://github.com/prebid/Prebid.js/blob/d5746c39c437b5e04505cdd75bdef3791eb28566/modules/consentManagementGpp.js#L380

You can mergeDeep regs, site, app, and user

Reply:
I updated it to always use the ortb2 data. For the regs we can not mergeDeep, our backend supports openRTB 2.5 and the gpp under "regs.ext.gpp".